src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2455,27 +2460,39 @@ public override BoundNode VisitMethodGroup(BoundMethodGroup node)
             return null;
         }
 
-        public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperator node)
+#nullable enable
+
+        public override BoundNode? VisitNullCoalescingOperator(BoundNullCoalescingOperator node)
         {
             VisitRvalue(node.LeftOperand);
             if (IsConstantNull(node.LeftOperand))
             {
-                VisitRvalue(node.RightOperand);
+                Visit(node.RightOperand);
             }
             else
             {
-                var savedState = this.State.Clone();
+                var savedState = node.LeftOperand is BoundConditionalAccess
```

Comment:
There might be a need to change this to handle certain BoundConversions here, but the specification doesn't currently indicate that a ?? operand which is a cast containing a conditional access should do anything differently than today. However, it may be a good idea to change that.

Reply:
I was unable to find scenarios where the LHS of a `??` is a BoundConversion where:
- there is no explicit cast
- there are no errors on the node
- Operand is a BoundConditionalAccess

Therefore, I put an assertion and a PROTOTYPE in the code for the moment.

When debugging the bootstrap failures I did experiment a fair bit with different scenarios to try and hit the assertion. If you have ideas for scenarios that would actually hit the assertion as written, that would be welcome.

When we start handling more kinds of containing expressions, I feel like the correct way to handle these conversion nodes is going to become more clear, so I'd like to punt it to the next PR (for `==`/`!=`) if that is acceptable.

Reply:
I would expect to see a conversion on the left side for a scenario like this:

```cs
C c = null;
D d = c ?? default(D);

class C {}

struct D
{
    public static implicit operator D(C c) => default;
}
```

Make the `c` variable a conditionally-access method with a null, and we should be able to observe behavior.

---
In reply to: [595506242](https://github.com/dotnet/roslyn/pull/51567#discussion_r595506242) [](ancestors = 595506242)